ethcore/src/block_queue.rs

```diff
@@ -285,18 +285,24 @@ impl BlockQueue {
 	}
 
 	/// Mark given block and all its children as bad. Stops verification.
-	pub fn mark_as_bad(&mut self, hash: &H256) {
+	pub fn mark_as_bad(&mut self, hashes: &[H256]) {
```

Comment:
should not clear what `hashes` means in this context. consider renaming to `bad_blocks` or `bad_hashes`, if that's what is meant.